superduperdb/backends/query_dataset.py

```diff
@@ -139,6 +147,11 @@ def __init__(
         self._max_cache_size = prefetch_size
         self._cache: ExpiryCache = self._fetch_cache()
         self._total_documents = self.count_documents()
+        self._map_funcs: t.List[t.Callable] = []
+
+    def map(self, func):
```

Comment:
Please explain.

Reply:
I revert this class and convert it to Dataset later to process the map function